core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/EmbeddedExpressionSpec.scala

```diff
@@ -0,0 +1,248 @@
+  // TODO: Make a named function to call out to "dateCompare()" (maybe a primitive algebra?)
+  //       Or just hack it for now.
+  // now.year - dob.year - (if (now.dayOfYear < (dob.dayOfYear + (if (dob.isLeapYear) 1 else 0)) 1 else 0)
+  lazy val ageFromDateOfBirth: Expr.Definition[Unit] = {
+    val now = LocalDate.now()
+    define(FactTypes.Age) {
+      withFactsOfType(FactTypes.DateOfBirth).where { facts =>
+        facts.map { fact =>
+          fact.get(_.select(_.value).select(_.getYear)).subtractFrom(now.getYear)
+        }
+      }
+    }
+  }
+
+  lazy val isOver18: RootExpr[Boolean, Unit] = {
+    usingDefinitions(ageFromDateOfBirth) {
+      withFactsOfType(FactTypes.Age).where {
+        _.exists {
+          _.get(_.select(_.value)) >= 18
+        }
+      }
+    }
+  }
+
+  "compute age correctly" in {
+    val result = eval(JoeSchmoe.factTable) {
+      usingDefinitions(ageFromDateOfBirth) {
+        withFactsOfType(FactTypes.Age).returnInput
+      }
+    }
+    val ages = result.output.value.map(_.value)
+    ages should contain only JoeSchmoe.age.value
+  }
+
+  "overwrite another fact type" in {
+    val dob = FactTypes.DateOfBirth(LocalDate.of(1990, 1, 1))
+    val facts = FactTable(
+      List(
+        FactTypes.Age(15),
+        dob,
+      ),
+    )
+    val result = eval(facts)(isOver18)
+    assert(result.output.value)
+    pendingUntilFixed {
+      assertResult(FactsMatch(Facts(dob)))(result.output.evidence)
+    }
+  }
+
+  "embedding an expression inside a logical operator inside a 'withFactsOfType'" when {
+
+    def insideProbOfWeightloss(cond: Expr[Id, Double, Boolean, Unit]): RootExpr[Boolean, Unit] = {
+      withFactsOfType(FactTypes.ProbabilityToUse).where {
+        _.flatMap {
+          _.getFoldable[List, Double] {
+            _.select(_.value).select(_.scores).atKey("weightloss").to(List)
+          }
+        }.exists {
+          _.embed(cond)
+        }
+      }
+    }
+
+    def weightMeasuredWithin(window: Window[Int]): RootExpr[Boolean, Unit] = {
+      import cats.syntax.invariant._
+      val doubleWindow = window.imap(_.toDouble)(_.toInt)
+      withFactsOfType(FactTypes.WeightMeasurement).where {
+        _.exists {
+          _.get(_.select(_.value).select(_.value)).within(doubleWindow)
+        }
+      }
+    }
+
+    val trueEmbedded = weightMeasuredWithin(Window.greaterThan(200))
+    val falseEmbedded = weightMeasuredWithin(Window.greaterThan(300))
+    val embeddedFacts = Facts(JoeSchmoe.weight)
+
+    def trueLiteral: Expr[Id, Double, Boolean, Unit] = {
+      within(input, Window.greaterThan(0.7))
+    }
+
+    def falseLiteral: Expr[Id, Double, Boolean, Unit] = {
+      within(input, Window.greaterThan(0.9))
+    }
+
+    val literalFacts = Facts(JoeSchmoe.probs)
+
+    "inside an 'or'" should {
+
+      "return 'true' when embedding a 'true' expression BEFORE a 'true' literal" in {
+        val q = insideProbOfWeightloss(or(trueEmbedded, trueLiteral))
```

Comment:
It's a nitpick, but `query`  would make the eval line read `val result = eval(JoeSchmoe.factTable)(query)`.